app/models/space.rb

```diff
@@ -92,11 +94,21 @@ def cleaned_graph
     {'metrics' => Array.wrap(cleaned_metrics), 'guesstimates' => Array.wrap(cleaned_guesstimates)}
   end
 
+  def visible_to?(user)
+    is_public? || user.id == self.user_id || user.member_of?(self.organization_id)
+  end
```

Comment:
Yes, realizing there may be some immediate refactors here already with other auth checks performed by the spaces controller; should look into that.